### PR TITLE
Multiple questions

### PR DESCRIPTION
**Pull request by liyiy**
Support multiple questions per survey

https://github.com/PostHog/feature-surveys/assets/25164963/2ea1aafc-3b04-4f26-8b4d-1e3a769ba33b

Multiple questions response will be returned like so:

Screen Shot 2023-09-28 at 11 23 40 AM

To click test locally:
Replace `getActiveMatchingSurveys` with the below:
```
    const multipleQuestions = [{
        name: 'multiple-questions-surv',
        id: 'a1b2c3d4',
        questions: [
            { question: 'please explain your response?', type: 'open', required: false },
            { question: 'why did you decide to use posthog?', description: 'hello', type: "multiple_choice", choices: ["affordable pricing", "simple to use", "features variety", "improved company's product analytics"] },
            { question: 'how satisfied are you?', type: 'rating', display: 'emoji', scale: 5 },
            { question: 'click on this link', type: 'link', link: 'https://posthog.com' },
        ]
    }].forEach((survey) => {
```

### CODE REVIEW

File: site.ts [hunk 1]
```diff
@@ -333,83 +336,82 @@ export const callSurveys = (posthog, forceReload = false) => {
...
-                                })
-                            }
-                            const countdownEl = thankYouElement.querySelector('.thank-you-message-countdown');
-                            if (countdownEl) {
-                                let count = 3;
+            if (!localStorage.getItem(`seenSurvey_${survey.id}`)) {
+                const shadow = createShadow(style(survey.id, survey?.appearance), survey.id)
+                let surveyPopup
+                if (survey.questions.length < 2) {
+                    surveyPopup = createSingleQuestionSurvey(posthog, survey, survey.questions[0])
```
Comment (liyiy) [inline site.ts:360]: this block is the only new stuff in the `callSurveys` method

**Comment by neilkakkar** [comment]
Not a big fan of answers being in an array like this, since it further complicates how to display results, and is a bit awkward for questions where there isn't really a response, like "link clicked".

I'd prefer sticking to what we decided earlier for this schema (which has its own awkwardness but I feel works much better for result querying & having empty responses in between): `$survey_response` is for 0th index question, `$survey_response_1` for 1st index question, and so on.

Has something changed since we decided that^? -> https://github.com/PostHog/feature-surveys/pull/3#discussion_r1247737413

**Comment by neilkakkar** [comment]
Also because we need to be backwards compatible with the current schema, and turning questions/responses into an array can work, although isn't clear enough imo because that further leads to nested arrays etc. etc.

**Comment by liyiy** [comment]
> Not a big fan of answers being in an array like this, since it further complicates how to display results, and is a bit awkward for questions where there isn't really a response, like "link clicked".
> 
> I'd prefer sticking to what we decided earlier for this schema (which has its own awkwardness but I feel works much better for result querying & having empty responses in between): `$survey_response` is for 0th index question, `$survey_response_1` for 1st index question, and so on.
> 
> Has something changed since we decided that^? -> [#3 (comment)](https://github.com/PostHog/feature-surveys/pull/3#discussion_r1247737413)

Easy fix! Just wanted to push something out quickly and forgot :)

**Pull request merged**